### PR TITLE
[CDAP-20893] Move created workload identities to system namespace

### PR DESCRIPTION
**Pull request by dli357**

### CODE REVIEW

File: cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java
Comment (itsankit-google) [inline cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java]: This comparison looks incorrect, `namespace` will always be `SYSTEM` here right?

We need to check the actual namespace from where the request was triggered in task worker.

Comment (itsankit-google) [reply cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java]: For example, in HDF where we don't set `k8s.namespace` property in `default` namespace but identities are present in `system` namespace, this condition will return true.

**Review by itsankit-google — COMMENTED** [review]
should we also make the below request also of type `POST`?https://github.com/cdapio/cdap/blob/05cdf8148a2a8dec035eb7a748278d39c43770ba/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/namespace/credential/handler/GcpWorkloadIdentityHttpHandlerInternal.java#L68

File: cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java
Comment (dli357) [reply cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java]: Good catch. I addd a new property field to encapsulate this value.

**Comment by dli357** [comment]
> should we also make the below request also of type `POST`?
> 
> https://github.com/cdapio/cdap/blob/05cdf8148a2a8dec035eb7a748278d39c43770ba/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/namespace/credential/handler/GcpWorkloadIdentityHttpHandlerInternal.java#L68

We could, but the main reason I made the other one POST is that we need to encapsulate the entire CredentialProvisionContext as data, since scope are not necessarily a core abstraction of the CredentialProvider. However, for GcpWorkloadIdentityHttpHandlerInternal, the contract does allow scopes because GCP tokens support scope.

If the passed data class becomes more context, we could consider changing the contract of GcpWorkloadIdentityHttpHandlerInternal too.

**Review by itsankit-google — APPROVED** [review]
LGTM, please verify once after deploying CDAP on GKE if everything works fine before merging.

File: cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java
Comment (itsankit-google) [inline cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java:218]: `cdapWrappedNamespace` and not `k8sNamespace` which will be default always for non-hybrid mode.

File: cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/ProvisionedCredentialCacheKey.java
Comment (itsankit-google) [inline cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/ProvisionedCredentialCacheKey.java]: Please also add identity in hashcode computation.

File: cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java
Comment (itsankit-google) [reply cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java:218]: Discussed offline to include identity in cache key.

File: cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/ProvisionedCredentialCacheKey.java
Comment (dli357) [reply cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/ProvisionedCredentialCacheKey.java]: Done, thanks!

**Pull request merged**
